### PR TITLE
Fixes a number of null loc playsound assert fails

### PR DESCRIPTION
**Pull request by gbasood**
The causes I addressed in this are:
- Telesci's awful spark() proc, which was the number one cause of assert fails in the logs I checked.
- Rogue drone event end
- Igniters in bombs causing sparks AFTER their assembly blows up
- Gold slime core + plasma spawning hivebots without immediately giving  them a location

I tested the gold slime core reaction and telesci. Forgot to test the bomb thing.

### CODE REVIEW

**Comment by ihadtoregisterforthis** [comment]
checked

Holey moley that's a lot of issues and PRs lately

**Comment by Exxion** [comment]
Yeah, `spark()` was the next big one after the other assert fails I fixed awhile ago, but I never did manage to figure out exactly what was breaking it

Also nice #9000

**Comment by Intigracy** [comment]
You could fix the meteor/flash one too while you're at it

**Comment by Shadowmech88** [comment]
I got one while firing flares from the roulette revolver in #8710.
I fixed it by just adding a check to verify that the loc wasn't null before the ASSERT in sound.dm.
Is that a bad way to fix it?

**Comment by Exxion** [comment]
Yes
That's the whole point of the ASSERT
ASSERT is a debug tool that runtimes if the value it is provided with is falsy, so putting a check before it defeats its entire purpose

**Comment by gbasood** [comment]
There's some situations where the only way to fix is a null loc check before playsound, but they're VERY few and far between.

What you should ACTUALLY be doing is figuring out why that thing is trying to play a sound at a null loc. Because more often than not, what you're actually encountering is something getting nullspaced before it hits that code block, but before the proc (or the proc that calls it) fires.

File: code/modules/events/rogue_drones.dm [hunk 1]
```diff
@@ -38,14 +38,16 @@
 /datum/event/rogue_drone/end()
 	var/num_recovered = 0
 	for(var/mob/living/simple_animal/hostile/retaliate/malf_drone/D in drones_list)
```
Comment (clusterfack) [inline code/modules/events/rogue_drones.dm]: Is this spawning drones in nullspace?

Comment (clusterfack) [inline code/modules/events/rogue_drones.dm]: Nevermind its despawning them so much for garbage collection of these

Comment (gbasood) [inline code/modules/events/rogue_drones.dm:41]: It's despawning them by calling qdel() yeah, I changed it from the broken code that existed before that I think made them not get qdel'd properly.

Comment (clusterfack) [inline code/modules/events/rogue_drones.dm]: Yeah qdel wont save them considering drones list holds a reference to them

File: code/modules/events/rogue_drones.dm [hunk 2]
```diff
@@ -38,14 +38,16 @@
...
-		sparks.start()
+		var/locc = get_turf(D)
+		if(locc)
+			var/datum/effect/effect/system/spark_spread/sparks = new /datum/effect/effect/system/spark_spread()
+			sparks.set_up(3, 0, locc)
+			sparks.start()
 		D.z = 2
 		D.has_loot = 0
 
+		drones_list -= D
```
Comment (Intigracy) [inline code/modules/events/rogue_drones.dm]: :+1:

Comment (clusterfack) [inline code/modules/events/rogue_drones.dm]: Is that an indirect reply to my comment? I'm talking about if a drone is destroyed during the time in which this event is occurring.

File: code/modules/events/rogue_drones.dm (hunk 1 above)
Comment (gbasood) [inline code/modules/events/rogue_drones.dm:41]: And now drones remove themselves from the event's drone list in their Destroy()

File: code/modules/events/rogue_drones.dm (hunk 2 above)
Comment (Intigracy) [inline code/modules/events/rogue_drones.dm]: No, I'm giving that a thumbs up.

**Pull request merged**